pkg/apis/eventsource/v1alpha1/types.go

```diff
@@ -1059,46 +1059,54 @@ type BitbucketBasicAuth struct {
 
 // BitbucketServerEventSource refers to event-source related to Bitbucket Server events
 type BitbucketServerEventSource struct {
-	// Webhook holds configuration to run a http server
+	// Webhook holds configuration to run a http server.
 	Webhook *WebhookContext `json:"webhook,omitempty" protobuf:"bytes,1,opt,name=webhook"`
-	// DeprecatedProjectKey is the key of project for which integration needs to set up
-	// Deprecated: use Repositories instead. Will be unsupported in v1.8
+	// DeprecatedProjectKey is the key of project for which integration needs to set up.
+	// Deprecated: use Repositories instead. Will be unsupported in v1.8.
 	// +optional
 	DeprecatedProjectKey string `json:"projectKey,omitempty" protobuf:"bytes,2,opt,name=projectKey"`
-	// DeprecatedRepositorySlug is the slug of the repository for which integration needs to set up
-	// Deprecated: use Repositories instead. Will be unsupported in v1.8
+	// DeprecatedRepositorySlug is the slug of the repository for which integration needs to set up.
+	// Deprecated: use Repositories instead. Will be unsupported in v1.8.
 	// +optional
 	DeprecatedRepositorySlug string `json:"repositorySlug,omitempty" protobuf:"bytes,3,opt,name=repositorySlug"`
-	// Repositories holds a list of repositories for which integration needs to set up
+	// Projects holds a list of projects for which integration needs to set up, this will add the webhook to all repositories in the project.
+	// +optional
+	Projects []string `json:"projects,omitempty" protobuf:"bytes,4,opt,name=projects"`
+	// Repositories holds a list of repositories for which integration needs to set up.
 	// +optional
-	Repositories []BitbucketServerRepository `json:"repositories,omitempty" protobuf:"bytes,4,rep,name=repositories"`
+	Repositories []BitbucketServerRepository `json:"repositories,omitempty" protobuf:"bytes,5,opt,name=repositories"`
```

Comment:
It should be `rep`.

Reply:
Should `Projects []string `json:"projects,omitempty" protobuf:"bytes,4,opt,name=projects"` also be `rep` as well since it is a list of values as well?

Reply:
Done ✅